p2pserver/link/link.go

```diff
@@ -144,8 +144,15 @@ func unpackNodeBuf(this *Link, buf []byte) {
 			this.CloseConn()
 			return
 		}
-
-		rxBuf.len = types.PayloadLen(rxBuf.p)
+		payloadLen, err := types.PayloadLen(rxBuf.p)
+		if err != nil {
+			rxBuf.p = nil
+			rxBuf.len = 0
+			log.Warn("get payload len error")
+			this.CloseConn()
+			return
+		}
+		rxBuf.len = payloadLen
```

Comment:
why not check MAX_MSG_LEN here?
if payloadLen > max_msg_len, close the connection?

Reply:
if payloadLen > max_msg_len, the PayloadLen function will return a error.

Reply:
Great. 
Then is it still necessary to checking MAX_MSG_LEN in the deserialization of all kinds of payload?
The payload length will never larger than max_msg_len.

Reply:
In this case, the msg received from link will be restricted of length. And if somewhere other call the deserialization function, it should do the same logic. So the checking inside the deserialization is a protect.

Reply:
ok